### PR TITLE
Move some common Go stuff up into a new 'ct' package.

### PR DESCRIPTION
**Pull request by AlCutter**
Also fixes a few linty errors while I'm at it.

### CODE REVIEW

File: go/client/logclient.go [hunk 1]
```diff
@@ -271,10 +268,10 @@ func (c *LogClient) GetSTH() (sth *SignedTreeHead, err error) {
 	return
 }
 
-// Attempts to retrieve the entries in the sequence [|start|, |end|] from the CT
+// GetEntries attempts to retrieve the entries in the sequence [|start|, |end|] from the CT
```
Comment (pphaneuf) [inline go/client/logclient.go]: You going to have to decide whether you wrap, or not. :stuck_out_tongue:

**Comment by pphaneuf** [comment]
LGTM

File: go/client/logclient.go (hunk 1 above)
Comment (AlCutter) [inline go/client/logclient.go]: Apparently comment blocks 'typically' wrap at 80 for readability in the code, even though the code doesn't and godoc will sort it all out anyway.

File: go/client/logclient.go [hunk 2]
```diff
@@ -52,11 +49,11 @@ type addChainRequest struct {
...
-	SCTVersion Version `json:"sct_version"` // SCT structure version
-	ID         string  `json:"id"`          // Log ID
-	Timestamp  uint64  `json:"timestamp"`   // Timestamp of issuance
-	Extensions string  `json:"extensions"`  // Holder for any CT extensions
-	Signature  string  `json:"signature"`   // Log signature for this SCT
+	SCTVersion ct.Version `json:"sct_version"` // SCT structure version
+	ID         string     `json:"id"`          // Log ID
+	Timestamp  uint64     `json:"timestamp"`   // Timestamp of issuance
+	Extensions string     `json:"extensions"`  // Holder for any CT extensions
+	Signature  string     `json:"signature"`   // Log signature for this SCT
```
Comment (benlaurie) [inline go/client/logclient.go]: Doesn't json name the fields automatically?

**Pull request merged**